### PR TITLE
Public file view

### PR DESCRIPTION
**Pull request by felixheidecke**
Allow media to be viewed in public links.

Fixes https://github.com/owncloud/files_mediaviewer/issues/17
Fixes https://github.com/owncloud/files_mediaviewer/issues/37

Closes https://github.com/owncloud/files_mediaviewer/pull/39

### CODE REVIEW

**Comment by felixheidecke** [comment]
Woops, on to much. Will reverse

**Comment by PVince81** [comment]
still not encoded:
image

File: src/scripts/Viewer.vue [hunk 1]
```diff
@@ -37,32 +37,65 @@ export default {
...
+					a: 1,
+					t : this.sharingToken
+				})
+			}
+
+			else {
+				path = OC.joinPaths(
+					OC.linkToRemoteBase('dav/files'),
+					OC.getCurrentUser().uid,
+					item.path,
```
Comment (PVince81) [inline src/scripts/Viewer.vue]: you need to encode these somehow.

I thought we had utility methods for that somewhere

Comment (felixheidecke) [reply src/scripts/Viewer.vue]: I am looking for those … didn't find any so far. I'll go ahead and look at what the video player did.

**Comment by PVince81** [comment]
and with the public endpoint, same problem like in the other PR with the double prefix:
image

File: src/scripts/Viewer.vue (hunk 1 above)
Comment (felixheidecke) [reply src/scripts/Viewer.vue]: Currently, path creation is super dirty. There is encoding in buildQueryString already, but joinPaths doesn't. Not sure, if I'm using all the wrong methods here.

**Comment by PVince81** [comment]
I found `OC.encodePath()` which will `encodeURIComponent()` each path section but leave the slashes.

am testing locally but strangely even with the encoded path, the request to the server seems troublesome. it does answer with 206 partial content...

investigating...

**Comment by PVince81** [comment]
WAIT, the problem isn't about encoding :face_with_head_bandage: 

The problem is about the **very first** video in the list: regardless what video it is, the player fails.
If I create an entry to have another video before the one I mentionned above, it works fine even without URL encoding...

I've raised this here: https://github.com/owncloud/files_mediaviewer/issues/37

I'll now test this PR again and skip the first entry.

**Comment by PVince81** [comment]
- [ ] BUG: the doubled "owncloud/owncloud" now appears only for picture viewing in public mode.

videos work fine

**Comment by felixheidecke** [comment]
`OC.generateUrl("")` and `OC.linkToRemoteBase("")` … vastly different names, yet they do the same thing either prepending .../index.php and .../remote.php. No wonder I'm getting confused :-P

**Comment by felixheidecke** [comment]
@PVince81 please retry … I figured the url/path generation out.

**Comment by PVince81** [comment]
- [ ] BUG: download button doesn't work

```
<img width="562" alt="image" src="https://user-images.githubusercontent.com/277525/52708731-3654d680-2f8b-11e9-90a2-435456966041.png">
```

it should be possible to get the download link from the FileList object.
you probably already do so for videos

**Comment by PVince81** [comment]
Everything else works fine.

@felixheidecke please fix the above and I think we're good to go

**Comment by felixheidecke** [comment]
Okay. Downloading in frontend works. I'll start writing tests asap to not run into regression or forget to test scenarios.

**Comment by PVince81** [comment]
download button works now

**Comment by felixheidecke** [comment]
@PVince81 can we merge it as "Public file view" is done and it closes a few issues.

**Pull request merged**